cli/cli_test.go

```diff
@@ -353,3 +358,83 @@ func Example_max_results() {
 	// quit
 	// node drained and shutdown: ok
 }
+
+// TestFlagUsage is a basic test to make sure the fragile
+// help template does not break.
+func TestFlagUsage(t *testing.T) {
+	defer leaktest.AfterTest(t)
+
+	// Override os.Stdout with our own.
+	old := os.Stdout
+	r, w, _ := os.Pipe()
+	os.Stdout = w
+
+	done := make(chan bool)
+	var buf bytes.Buffer
+	// copy the output in a separate goroutine so printing can't block indefinitely.
+	go func() {
+		// Copy reads 'r' until EOF is reached.
+		_, _ = io.Copy(&buf, r)
+		done <- true
+	}()
+
+	if err := Run([]string{"help"}); err != nil {
+		fmt.Println(err)
+	}
+
+	// back to normal state
+	w.Close()
+	_ = <-done
+	close(done)
```

Comment:
You don't need to close the channel on this side. And you don't need to assign the result to `_`. We tend to use `chan struct{}` for valueless channels.